src/Argu/PreCompute.fs

```diff
@@ -66,7 +65,13 @@ let mkPrimitiveParser (name : string) (parser : string -> 'T) (unparser : 'T ->
         UnParser = fun o -> unparser (o :?> 'T)
     }
 
-let primitiveParsers =
+let private dict (s: ('key * 'value) []) =
+    let result = System.Collections.Generic.Dictionary<'key, 'value>(s.Length)
+    for pair in s do
+        result.Add(fst pair, snd pair)
+    result :> System.Collections.Generic.IDictionary<_,_>
```

Comment:
why the need to upcast?

Reply:
It was to keep the signature the same but as the result is only ever used in private types it's useless. Fixed it.